staging/src/k8s.io/apimachinery/pkg/util/errors/errors.go

```diff
@@ -147,6 +147,28 @@ func Flatten(agg Aggregate) Aggregate {
 	return NewAggregate(result)
 }
 
+// UniqueAggregateWithCount removes duplicates from flattened Aggregate
+// and returns count for each unique error present in the Aggregate
+func UniqueAggregateWithCount(agg Aggregate) Aggregate {
+	if agg == nil || len(agg.Errors()) == 0 {
+		return agg
+	}
+	errs := Flatten(agg).Errors()
```

Comment:
If your concern is why I am calling Flatten, then the reason is that this function is not specific to scheduler and so the need to call Flatten. Also what @jayunit100 had done was for specific to the fit errors where these are other (atleast previously) unexpected errors.